### PR TITLE
Fixed #31061 -- Ignored positional args in django.urls.resolve() when all optional named parameters are missing.

### PR DESCRIPTION
**Pull request by felixxm**
Regression in 76b993a117b61c41584e95149a67d8a1e9f49dd1.

Thanks Claude Paroz for the report.

ticket-31061

### CODE REVIEW

File: docs/releases/3.0.1.txt [hunk 1]
```diff
@@ -13,3 +13,7 @@ Bugfixes
   inside Jupyter and other environments that force an async context, by adding
   and option to disable :ref:`async-safety` mechanism with
   ``DJANGO_ALLOW_ASYNC_UNSAFE`` environment variable (:ticket:`31056`).
+
+* Fixed a regression in Django 3.0 where :func:`~django.urls.re_path` passes
+  positional arguments to the view when all captured named groups are missing
+  (:ticket:`31061`).
```
Comment (carltongibson) [inline docs/releases/3.0.1.txt]:
```
* Fixed a regression in Django 3.0 where ``RegexPattern``, used by
  :func:`~django.urls.re_path`, would return positional arguments to be passed
  to the view when all optional named groups were missing (:ticket:`31061`).
```

**Comment by terencehonles** [comment]
I ran into something very similar. I wasn't sure if I should put my report here or if I should create an issue on the Django project website. I went with the Django project website, but mentioning it here to see what @felixxm and @carltongibson think: https://code.djangoproject.com/ticket/31069

**Comment by felixxm** [comment]
@terencehonles I left comment on the ticket-31069, I believe that the current behavior is correct.

@carltongibson Thanks for the review :+1: I updated PR.

File: docs/releases/3.0.1.txt [hunk 2]
```diff
@@ -13,3 +13,7 @@ Bugfixes
   inside Jupyter and other environments that force an async context, by adding
   and option to disable :ref:`async-safety` mechanism with
   ``DJANGO_ALLOW_ASYNC_UNSAFE`` environment variable (:ticket:`31056`).
+
+* Fixed a regression in Django 3.0 where ``RegexPattern``, used by
+  :func:`~django.urls.re_path`, returns positional arguments to be passed to
```
Comment (carltongibson) [inline docs/releases/3.0.1.txt]: `would return`?  i.e. It **used** to do this but we fixed it. (It's not current behaviour.)

?

Comment (felixxm) [reply docs/releases/3.0.1.txt]: I think we use present tenses in release notes ([an example](https://github.com/django/django/blob/master/docs/releases/2.2.2.txt#L40-L59)), i.e. Django 3.0 still behaves that way :thinking:

Comment (felixxm) [reply docs/releases/3.0.1.txt]: Updated.

**Pull request merged**